examples/pruning/xgboost_cv_integration.py

```diff
@@ -0,0 +1,69 @@
+"""
+Optuna example that demonstrates a pruner for XGBoost.cv.
+
+In this example, we optimize the validation auc of cancer detection using XGBoost.
+We optimize both the choice of booster model and their hyperparameters. Throughout
+training of models, a pruner observes intermediate results and stop unpromising trials.
+
+You can run this example as follows:
+    $ python xgboost_cv_integration.py
+
+"""
+
+import sklearn.datasets
+import xgboost as xgb
+
+import optuna
+
+
+def objective(trial):
+    train_x, train_y = sklearn.datasets.load_breast_cancer(return_X_y=True)
+    dtrain = xgb.DMatrix(train_x, label=train_y)
+
+    param = {
+        'silent': 1,
+        'objective': 'binary:logistic',
+        'eval_metric': 'auc',
+        'booster': trial.suggest_categorical('booster', ['gbtree', 'gblinear', 'dart']),
+        'lambda': trial.suggest_loguniform('lambda', 1e-8, 1.0),
+        'alpha': trial.suggest_loguniform('alpha', 1e-8, 1.0)
+    }
+
+    if param['booster'] == 'gbtree' or param['booster'] == 'dart':
+        param['max_depth'] = trial.suggest_int('max_depth', 1, 9)
+        param['eta'] = trial.suggest_loguniform('eta', 1e-8, 1.0)
+        param['gamma'] = trial.suggest_loguniform('gamma', 1e-8, 1.0)
+        param['grow_policy'] = trial.suggest_categorical('grow_policy', ['depthwise', 'lossguide'])
+    if param['booster'] == 'dart':
+        param['sample_type'] = trial.suggest_categorical('sample_type', ['uniform', 'weighted'])
+        param['normalize_type'] = trial.suggest_categorical('normalize_type', ['tree', 'forest'])
+        param['rate_drop'] = trial.suggest_loguniform('rate_drop', 1e-8, 1.0)
+        param['skip_drop'] = trial.suggest_loguniform('skip_drop', 1e-8, 1.0)
+
+    pruning_callback = optuna.integration.XGBoostPruningCallback(trial, 'test-auc')
+    history = xgb.cv(
+        param,
+        dtrain,
+        num_boost_round=100,
+        callbacks=[pruning_callback]
+    )
+
+    best_auc = history['test-auc-mean'].max()
```

Comment:
I put a comment for those lines.

Reply:
@smly Thanks for you comment, I followed your instruction. Is it ready to merge?

Reply:
Thanks for the update. LGTM:+1: